engine/engine.go

```diff
@@ -326,18 +333,23 @@ func (e *compatibilityEngine) NewRangeQuery(ctx context.Context, q storage.Query
 	}
 
 	return &compatibilityQuery{
-		Query:  &Query{exec: exec, opts: opts},
-		engine: e,
-		expr:   expr,
-		t:      RangeQuery,
+		Query:       &Query{exec: exec, opts: opts},
+		engine:      e,
+		expr:        expr,
+		t:           RangeQuery,
+		debugWriter: e.debugWriter,
 	}, nil
 }
 
 type ExplainableQuery interface {
 	promql.Query
 
 	Explain() *ExplainOutputNode
-	Profile()
+	Analyze() *AnalyzeOutputNode
+}
+type AnalyzeOutputNode struct {
+	OperatorTime *model.TimingInformation
```

Comment:
Let's name it a bit generically, so that we can add other stuff in the future.
```suggestion
	OperatorTelemetry *model.TimingInformation
```